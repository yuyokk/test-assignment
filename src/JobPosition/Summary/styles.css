.summary {
}
@media (min-width: 768px) {
  .summary {
    width: 570px;
  }
}

.summary__header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 15px;
}

.summary__header h1 {
  display: flex;
  align-items: center;
  margin-bottom: 0;
}

.summary__education-wrapper .tag {
  margin-right: 7.5px;
  margin-bottom: 7.5px;
}

.summary__education-wrapper .tag:last-child {
  margin-bottom: 0;
}
