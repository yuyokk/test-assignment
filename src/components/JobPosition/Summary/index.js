import React from 'react';
import PropTypes from 'prop-types';
import Panel from '../../_common/Panel';
import Badge from '../../_common/Badge';
import Button from '../../_common/Button';
import Icon from '../../_common/Icon';
import './styles.css';

function Summary(props) {
  const { eduLevels } = props;
  const educationLevels = [];

  for (const level in eduLevels) {
    if (eduLevels.hasOwnProperty(level)) {
      educationLevels.push(<Badge key={level} text={level} />);
    }
  }

  return (
    <Panel className="summary">
      <header className="summary__header">
        <h1>
          <Icon className="header-icon margin-right--md" type="success" /> Job criteria
        </h1>

        <Button className="summary__edit-btn" secondary onClick={() => props.onEdit()}>
          Edit
        </Button>
      </header>

      <p>
        <span className="text-dimmed">A minimum No. years of experience:</span>{' '}
        {props.yearsExperience}
      </p>

      <p>
        <span className="text-dimmed">No. of working hours (per week):</span> {props.minHours}-
        {props.maxHours} hours
      </p>

      <p>
        <strong>Level of education</strong>
      </p>

      <div className="summary__education-wrapper">{educationLevels}</div>
    </Panel>
  );
}

Summary.propTypes = {
  eduLevels: PropTypes.object,
  yearsExperience: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  minHours: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  maxHours: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onEdit: PropTypes.func.isRequired
};

export default Summary;
