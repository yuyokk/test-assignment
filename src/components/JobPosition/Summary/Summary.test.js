import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Summary from './index';

describe('<Summary />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Summary
          yearsExperience={1}
          eduLevels={{
            'level 1': true,
            'level 2': true
          }}
          minHours={32}
          maxHours={40}
          onEdit={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('handles onEdit click', () => {
    const onEditMock = jest.fn();
    const wrapper = mount(<Summary onEdit={onEditMock} />);
    wrapper
      .find('.summary__edit-btn')
      .first()
      .simulate('click');
    expect(onEditMock).toBeCalled();
  });
});
