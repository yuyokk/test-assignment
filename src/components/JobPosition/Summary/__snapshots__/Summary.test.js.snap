// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> renders correctly 1`] = `
<section
  className="summary panel"
>
  <header
    className="summary__header"
  >
    <h1>
      <span
        className="icon icon--success header-icon margin-right--md"
      />
       Job criteria
    </h1>
    <button
      className="summary__edit-btn btn btn-primary btn-secondary"
      onClick={[Function]}
    >
      Edit
    </button>
  </header>
  <p>
    <span
      className="text-dimmed"
    >
      A minimum No. years of experience:
    </span>
     
    1
  </p>
  <p>
    <span
      className="text-dimmed"
    >
      No. of working hours (per week):
    </span>
     
    32
    -
    40
     hours
  </p>
  <p>
    <strong>
      Level of education
    </strong>
  </p>
  <div
    className="summary__education-wrapper"
  >
    <span
      className="badge"
    >
      level 1
    </span>
    <span
      className="badge"
    >
      level 2
    </span>
  </div>
</section>
`;
